### PR TITLE
Fix global options with spaces marked by \OptionNotUsed, gh1238

### PR DESCRIPTION
**Pull request by muzimuzhi**
<s>`ltnews40.tex` not updated yet.</s>

The previous fix for #1238, PR #1239 was merged into `main` and released as part of [2023-11-01-PL1](https://github.com/latex3/latex2e/releases/tag/2023-11-01-PL1). Should this following fix be part of the next PL release too?

Also I found the previous fix has no ltnews paragraphs in ltnews38. PR #1239 once contained such update (https://github.com/latex3/latex2e/pull/1239/commits/a326d1a9cf4da4f0835aa5852ad5696fa7f24e89), but it was then reverted (https://github.com/latex3/latex2e/commit/c977ff3cc352c55c341a2ad4f4fc1206e9450b60). It seems ltnews38 is frozen for major updates, but I can only mention the previous fix in ltnews40. Is that so?

# Internal housekeeping

## Status of pull request

- Feedback wanted 
- Under development
- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [x] Rollback provided (if necessary)?
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

**Comment by josephwright** [comment]
Do we really need rollback here? Out-and-out bug so I'd say no.

**Comment by muzimuzhi** [comment]
> Do we really need rollback here? Out-and-out bug so I'd say no.

Reasonable. I once noticed that the previous fix #1239 didn't add rollback, but totally forgot about it when preparing this PR.

I'll remove rollback.

File: base/ltclass.dtx [hunk 1]
```diff
@@ -1515,7 +1515,9 @@
 %<latexrelease>\IncludeInRelease{2021/06/01}%
 %<latexrelease>                 {\OptionNotUsed}{filter unused option list}%
 %<*2ekernel|latexrelease>
-\def\@remove@eq@value#1=#2\@nil{#1}
+\ExplSyntaxOn
+\def\@remove@eq@value#1=#2\@nil{\tl_trim_spaces:n{#1}}
```
Comment (josephwright) [inline base/ltclass.dtx:1519]: Should this be expanded at this stage?

**Comment by muzimuzhi** [comment]
ltnews40 entry is added, which is based on @Skillmon's commit https://github.com/latex3/latex2e/pull/1239/commits/a326d1a9cf4da4f0835aa5852ad5696fa7f24e89.

File: base/ltclass.dtx (hunk 1 above)
Comment (muzimuzhi) [reply base/ltclass.dtx:1519]: It's the same as the similar treatment in `ltkeys.dtx`, which prohibits further expansion on key names, see the `\exp_not:n` in `\@@_remove_equals:w`:
https://github.com/latex3/latex2e/blob/22375e7b5e90cc9638f62806f69a11aa532430e5/base/ltkeys.dtx#L496-L498
https://github.com/latex3/latex2e/blob/22375e7b5e90cc9638f62806f69a11aa532430e5/base/ltkeys.dtx#L388-L390

Also from a previous discussion https://github.com/latex3/latex2e/issues/580#issuecomment-855303853, the agreement seemed to be that there is only a one-step expansion on the whole raw option list.

Comment (muzimuzhi) [reply base/ltclass.dtx:1519]: BTW the logic to remove optional `=value` part and trim spaces from remaining key name is currently implemented twice, once in `ltkeys.dtx` in pure expl3, and another one in `ltclass.dtx` in mixed 2e + expl3.

A follow-up code refactor (after this PR) can be done. Currently in building the LaTeX foramt, `ltclass.dtx` is loaded just before `ltkeys.dtx`.
https://github.com/latex3/latex2e/blob/22375e7b5e90cc9638f62806f69a11aa532430e5/base/format.ins#L210-L211

**Pull request merged**